python/src/equistore/operations/slice.py

```diff
@@ -0,0 +1,395 @@
+import warnings
+
+import numpy as np
+
+from ..tensor import TensorMap
+from ..block import TensorBlock
+from ..labels import Labels
+
+
+def slice(tensormap: TensorMap, samples_to_slice=None, properties_to_slice=None):
+    """
+    Slices an input :py:class:`TensorMap` along the samples and/or properties
+    dimension(s). `samples_to_slice` and `properties_to_slice` are
+    :py:class:`Labels` objects that specify the samples/properties
+    (respectively) names and indices that should be sliced, i.e. kept in the
+    output tensor.
+
+    Note that either `samples_to_slice` or `properties_to_slice`, or both,
+    should be specified as input. User warnings will be issued if any returned
+    blocks are empty after slicing.
```

Comment:
```suggestion
    should be specified as input.
```

this is already mentioned below